gui/wxpython/gcp/manager.py

```diff
@@ -1826,7 +1826,7 @@ def OnGeorect(self, event):
             else:
                 flags = "a"
 
-            busy = wx.BusyInfo(_("Rectifying images, please wait..."), parent=self)
```

Comment:
Strange, busy is used couple lines down there, was it reporting the warning for it?

Reply:
I tested out the tool to see what you commented here, but I couldn't get to the actual georectification part. I failed before that. Please see the error below. Could you please let me know what I did wrong? Additionally, I am wondering if the tool retains previous settings when the GUI is closed without clicking "Cancel". I don't know what to expect since I'm unfamiliar with this tool.
```
gcpmgr.InitMapDisplay()
  File "/usr/local/grass85/gui/wxpython/gcp/[manager.py](http://manager.py/)",
line 1275, in InitMapDisplay ColumnSorterMixin(self, ncols)
TypeError: ColumnSorterMixin.__init__() takes 2 positional arguments
but 3 were given
```
The source layer was `ortho_2001_t792_1m@user1` and the target layer was `Topographic OSM WMS - by terrestris` (WMS).
![g_gui_gcp_1](https://github.com/user-attachments/assets/175193f7-3ad0-46dd-b17a-822ee4e47a92)
![g_gui_gcp_2](https://github.com/user-attachments/assets/ef12ed18-ed9e-49f2-b45c-02f9986d003d)
![g_gui_gcp_3](https://github.com/user-attachments/assets/fde57a4a-4017-4ee9-bed7-f1258317f86f)
![g_gui_gcp_4](https://github.com/user-attachments/assets/08ab5335-8f38-4847-b9c4-2f58f845bd0e)

Reply:
All I was saying is that I don't understand why did you remove the `busy` variable, since it's actually used couple lines below, so I wondered if flake was reporting the F841 problem for it?

I don't think you needed to test it, but since you actually found a bug, could you please open an issue for it with the traceback?

Reply:
Flake8 complained of "busy" for F841, which is why I removed them. Flake8 doesn't pick up the same variables at once, in this case line 1829 "busy" appears after deleting line 1889.
```
mshukun@grads2213:~/Documents/GitHubProjects/grass/gui/wxpython/gcp$ flake8 --select=F841 manager.py
manager.py:1889:17: F841 local variable 'busy' is assigned to but never used
mshukun@grads2213:~/Documents/GitHubProjects/grass/gui/wxpython/gcp$ flake8 --select=F841 manager.py
manager.py:1829:13: F841 local variable 'busy' is assigned to but never used
```
For the tool bag, I will open an issue as you suggested.

Reply:
Looks like we can use [wx.BusyInfo with a context manager](https://docs.wxpython.org/wx.BusyInfo.html):
```
with wx.BusyInfo("Please wait, working..."):
    for i in range(10000):
    DoACalculation()
```

Could you please try to change that this was, which will avoid the the flake error.

Reply:
Please see "Solution 1". Does it looks correct to you? 

Alternatively, we can use "Solution: 2".
The "self" can also be used (e.g., self.busy, self._busy, self._ and etc.). In this way, they can remain formatted as they were originally. F841 issues are not raised. This was noticed because a few files contained "self.busy" and "self._busy" which weren't problematic.

Let me know your preference.

```python
# Solution 1:  
with wx.BusyInfo(
      _("Rectifying vector map <%s>, please wait...") % vect, parent=self
  ):
      wx.GetApp().Yield()

      ret, msg = RunCommand(
          "v.rectify",
          parent=self,
          getErrorMsg=True,
          quiet=True,
          input=vect,
          output=self.outname,
          group=self.xygroup,
          order=self.gr_order,
          overwrite=overwrite,
      )
```
```python
# Solution: 2
# self.busy, self._busy, self._ and etc.. 
  self._busy = wx.BusyInfo(
      _("Rectifying images, please wait..."), parent=self
  )
  wx.GetApp().Yield()

  ret, msg = RunCommand(
      "i.rectify",
      parent=self,
      getErrorMsg=True,
      quiet=True,
      group=self.xygroup,
      extension=self.extension,
      order=self.gr_order,
      method=self.gr_method,
      flags=flags,
      overwrite=overwrite,
  )

  del self._busy
```

Reply:
solution 1 is better